libvirt/tunnel/tunnel.sh

```diff
@@ -1,98 +1,130 @@
 #!/bin/bash
+# Load profile vars
+. profile.sh
 
 set -o nounset
 set -o pipefail
 set -o errexit
 
 # Number of clusters that can run on this host
-cluster_capacity=5
+echo "$CLUSTER_CAPACITY"
 
 # Bastion name & credentials
-bastion_name="bastion-z"
-token=""
+echo "$ENVIRONMENT"
+environment=$ENVIRONMENT
+token=$TOKEN
 
 # Default Ports
 libvirt_port=16509
 api_port=6443
 http_port=80
 https_port=443
 
-# Cluster id is added to bastion ports so that they don't conflict with each other
-cluster_id=1
-declare -A BASTION_PORTS
-BASTION_PORTS["LIBVIRT"]=$(($libvirt_port + $cluster_id))
-BASTION_PORTS["API"]=$(($api_port + $cluster_id))
-BASTION_PORTS["HTTP"]=$(($http_port + $cluster_id))
-BASTION_PORTS["HTTPS"]=$(($https_port + $cluster_id))
-
-# Bastion ports are calculated to ensure uniqueness, but the first env breaks the pattern
-declare -A CLUSTER_PORTS
-CLUSTER_PORTS["0,API"]=${BASTION_PORTS["API"]}
-CLUSTER_PORTS["0,HTTP"]=$((${BASTION_PORTS["HTTP"]} + 8000))
-CLUSTER_PORTS["0,HTTPS"]=$((${BASTION_PORTS["HTTPS"]} + 8000))
-for (( i=1; i<$cluster_capacity; i++ ))
-do
-  CLUSTER_PORTS["$i,API"]=$((${BASTION_PORTS["API"]} + 10000 * $i))
-  CLUSTER_PORTS["$i,HTTP"]=$((${BASTION_PORTS["HTTP"]} + 10000 * $i))
-  CLUSTER_PORTS["$i,HTTPS"]=$((${BASTION_PORTS["HTTPS"]} + 10000 * $i))
-done
```

Comment:
This block is probably OK to use as is.
I would probably rename CLUSTER_PORTS to BASTION_PORTS, and remove the second declare. This is because, these will never conflict.

The other thing you will want, is a LOCAL_PORTS array, that somehow builds the values that are currently passed as the local ports.